src/Resources/config/services/event_listeners.xml

```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<container xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://symfony.com/schema/dic/services"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd"
+>
+    <services>
+        <service id="Setono\SyliusRedirectPlugin\Menu\AdminMenuListener">
+            <tag name="kernel.event_listener" event="sylius.menu.admin.main" method="addAdminMenuItems"/>
+        </service>
+
+        <service id="Setono\SyliusRedirectPlugin\EventListener\NotFoundSubscriber">
+            <argument type="service" id="setono_sylius_redirect.manager.redirect"/>
+            <argument type="service" id="sylius.context.channel"/>
+            <argument type="service" id="setono_sylius_redirect.resolver.redirection_path"/>
+            <tag name="kernel.event_subscriber"/>
+        </service>
+
+        <service id="Setono\SyliusRedirectPlugin\EventListener\AbstractTranslationUpdateListener" abstract="true">
+            <argument type="service" id="validator"/>
+            <argument type="service" id="doctrine"/>
+            <argument type="service" id="setono_sylius_redirect.finder.removable_redirect"/>
+            <argument type="service" id="setono_sylius_redirect.factory.redirect"/>
+            <argument type="service" id="setono.sylius_redirect.decider.automatic_redirect_creation"/>
+            <argument />
+            <argument />
+        </service>
+
+        <service id="Setono\SyliusRedirectPlugin\EventListener\ProductUpdateListener"
+                 parent="Setono\SyliusRedirectPlugin\EventListener\AbstractTranslationUpdateListener">
+            <argument index="5">%setono_sylius_redirect.form.type.redirect.validation_groups%</argument>
```

Comment:
If you omit `<argument />`s at `AbstractTranslationUpdateListener `, you may not specify `index` at child service's arguments.

Reply:
Oh yes indeed.